design/Implemented/velero-uploader-configuration.md

```diff
@@ -102,25 +119,63 @@ Velero can set the `MaxParallelFileReads` parameter for Kopia's upload policy as
 
 ```go
 curPolicy := getDefaultPolicy()
-if uploaderCfg.ParallelFilesUpload > 0 {
-    curPolicy.UploadPolicy.MaxParallelFileReads = newOptionalInt(uploaderCfg.ParallelFilesUpload)
+if parallelUpload > 0 {
+	curPolicy.UploadPolicy.MaxParallelFileReads = newOptionalInt(parallelUpload)
 }
 ```
 
 #### Restic Parallel Upload Policy
 As Restic does not support parallel file upload, the configuration would not take effect, so we should output a warning when the user sets the `ParallelFilesUpload` value by using Restic to do a backup.
 
 ```go
-if uploaderCfg.ParallelFilesUpload > 0 {
-		log.Warnf("ParallelFilesUpload is set to %d, but Restic does not support parallel file uploads. Ignoring", uploaderCfg.ParallelFilesUpload)
+if parallelFilesUpload > 0 {
+	log.Warnf("ParallelFilesUpload is set to %d, but Restic does not support parallel file uploads. Ignoring", parallelFilesUpload)
 	}
 ```
 
 Roughly, the process is as follows: 
 1. Users pass the ParallelFilesUpload parameter and its value through the Velero CLI. This parameter and its value are stored as a sub-option within UploaderConfig and then placed into the Backup CR. 
 2. When users perform file system backups, UploaderConfig is passed to the PodVolumeBackup CR. When users use the Data-mover for backups, it is passed to the DataUpload CR. 
-3. Each respective controller within the CRs calls the uploader, and the ParallelFilesUpload from UploaderConfig in CRs is passed to the uploader. 
+3. The configuration will be stored in map[string]string type of field in CR.
+3. Each respective controller within the CRs calls the uploader, and the ParallelFilesUpload from map in CRs is passed to the uploader.
 4. When the uploader subsequently calls the Kopia API, it can use the ParallelFilesUpload to set the MaxParallelFileReads parameter, and if the uploader calls the Restic command it would output one warning log for Restic does not support this feature.
 
+### Sparse Option For Kopia & Restic Restore
+In many system files, there are numerous zero bytes or empty blocks that still occupy physical storage space. Sparse backup employs a more intelligent approach by only backing up the actual data-containing portions. For those empty blocks or zero bytes, it merely records their presence without actually storing them. This can significantly reduce the storage space required for backups, especially in situations where there is a substantial amount of empty data in large file systems.
```

Comment:
The current sparse option is for restore, not backup. For restore, the optimization is something like below:
If the backed up data is all zero, skip writing the block mapping the corresponding offset-length.

Reply:
Therefore, either we remove this explanation, or we add the correct explanation for restore.

Reply:
updated